File: java/MASTER_SLAVE_WHOLE_PART_PROXY/GestorDatosFuncion.java
public class GestorDatosFuncion {
    //Atributos
    private static DatosFuncion datosFuncion = new DatosFuncion();

    //Constructor
    public GestorDatosFuncion(){}

    //Métodos
    static DatosFuncion preparaDatosFuncion(String asiento, int dia, int mes, int anio, String pelicula){
        datosFuncion.setDatosFuncion(asiento, dia, mes, anio, pelicula);
        return datosFuncion;
    }
}

File: php/MASTER_SLAVE_WHOLE_PART_ABSTRACT_FACTORY/gestores/GestorDatosFuncion.php
<?php
require './datos/DatosFuncion.php';
class GestorDatosFuncion{
    //attr
    public static $datosFuncion;

    public static function init(){
        self::$datosFuncion = new DatosFuncion(); 
    }

    public static function preparaDatosFuncion($asiento, $dia, $mes,$anio, $pelicula){
        self::init();
        self::$datosFuncion->setDatosFuncion($asiento, $dia, $mes,$anio, $pelicula);
        return self::$datosFuncion;
    }
}